### DISCUSSION TITLE
Quantification of the native executable size

### DISCUSSION THREAD

**Discussion by Jiehong** [remkop/picocli#1909] [General]
Hello there,

Are there any study/blog/article about the impact of picocli on the size of the generated executable with native-image (GraalVM)?

At first glance, having only 1 single source file might not be helping for that, but then GraalVM should correctly identify what's used from what's not used.

I planned on trying to see the impact of picocli on the native executable size, but if someone has any info, that would be nice.

Also, would the programmatic API of picocli have any impact as well?

**Comment by remkop** [comment]
To be honest this may be a better question to ask the GraalVM project. I honestly don't know.

**Comment by Jiehong** [comment]
Answer from graal seems to indicate that this doesn't matter much: https://github.com/oracle/graal/discussions/5718

**Reply by remkop** [reply]
That is good to know!
